query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/commons/ServerGuesser.java

```diff
@@ -126,23 +131,45 @@ public static ServerGuesser buildGuesser(Connection connection) {
   }
 
   public List<String> getGroupObjectClass() {
-    return ImmutableList.of("groupOfNames", "group", "posixGroup");
+    return new ArrayList<>(
+        getResults("(|(objectClass=group)(objectClass=group*)(objectClass=*Group*))")
+            .stream()
+            .flatMap(entry -> Entries.getObjectClasses(entry).stream())
+            .filter(GROUP)
+            .map(ObjectClass::getNameOrOID)
+            .collect(Collectors.toCollection(TreeSet::new)));
   }
 
   public List<String> getGroupAttributeHoldingMember() {
-    return ImmutableList.of(MEMBER, "uniqueMember", "memberUid");
+    List<SearchResultEntry> queryResults =
+        getResults("(|(objectClass=group)(objectClass=group*)(objectClass=*Group*))");
+    Set<String> memberAttribute = new TreeSet<>();
+    for (SearchResultEntry entry : queryResults) {
+      for (Attribute attr : entry.getAllAttributes()) {
+        String name = attr.getAttributeDescription().getAttributeType().getNameOrOID();
+        if (name.toLowerCase().contains(MEMBER)) {
```

Comment:
❓ I'm concerned here and with the group OC method above that we'll be letting in attributes/OCs that should not be included. Random things like "memberCreationDate", for example.

Reply:
I understand the concern and it is a possibility.  I was attempting to make the code more accurate in its configuration guessing by supplying only attributes that actually existed rather than just possible ones.  I could create a filter such that only existing ones that match the filter would be allowed (basically the list before).  Or, if it is not worth it, I can revert it.  It just feels better to only see options that are valid in a particular setup.